gluten-substrait/src/main/scala/org/apache/gluten/extension/columnar/enumerated/RasOffload.scala

```diff
@@ -131,6 +134,9 @@ object RasOffload {
                   // TODO: Tag the original plan with fallback reason. This is a non-trivial work
                   //  in RAS as the query plan we got here may be a copy so may not propagate tags
                   //  to original plan.
+                  if (FallbackTags.maybeOffloadable(from)) {
+                    outComes.foreach(FallbackTags.add(from, _))
```

Comment:
I thought this practice may not work correctly for RAS for the reason commented above. Have you tested? Could delay on supporting RAS if not.